iceoryx_examples/small_memory/README.md

```diff
@@ -0,0 +1,76 @@
+# icehello with minimal memory usage
+
+## Introduction
+
+In some cases, it is needed to run iceoryx on systems with limited memory
+availability. For such cases, it is possible to configure the resource
+management limits as documented in the
+[configuration guide](./doc/website/advanced/configuration-guide.md).
+
+This configuration has two parts:
+* The resource limits set in cmake during configuration with options in the
+format `IOX_*`. This will influence the management shared memory segment size.
+* The memory pool configuration, set either in source code when creating the
+RouDi object, or when starting `iox-roudi` by specifying a TOML configuration
+file. This will influence the individual memory pool shared memory segment
+sizes.
+
+This is an example guide showing how to run the basic hello world example with
+truly minimal resource limits to achieve shared memory usage of less than 100
+kB.
+
+## How to build
+
+Build iceoryx RouDi application and the hello world example with the given
+CMake option configuration for minimal memory usage. For example, as follows:
+
+```
+cmake -Bbuild -Hiceoryx_meta -Ciceoryx_examples/small_memory/options.cmake -DEXAMPLES=ON
+cmake --build build --target iox-roudi iox-cpp-publisher-helloworld iox-cpp-subscriber-helloworld
+```
+
+Please note that it might be impossible to build some other iceoryx examples,
+since the decreased maximum length of service and runtime name strings might
+cause some static assertion failures. This problem demonstrates that the user
```

Comment:
@JakubSosnovec  These assertions are because the strings used in the other examples are too long then? Would it make sense to ensure all examples are compatible with the small_memory setup? Compiling maybe would be possible but we would get runtime problem as the resource limits are too small, I guess. So if we would like to ensure that all examples work, this would lead to bad compromises, e.g. having to increase the numbers again a bit because mempools are too small or not enough subscribers are possible. What do you think?

Reply:
@MatthiasKillat 

> Maybe it is worth considering to have a minimal with minimal string size that builds only one example?

Thats exactly what I did. The `small_memory` example that I added is basically just a guide on how to configure iceoryx just enough to be able to build the `icehello` example. I do know that other examples will fail, since some of them use very long strings.